### ISSUE TITLE
Cannot obtain call result from plugin compiled by tinygo with go-pdk

### ISSUE THREAD

**Issue by Celend** [extism/js-sdk#79]
- go version
  - 1.22.5 linux/amd64
- tinygo version
  - 0.32.0 linux/amd64 (using go version go1.22.5 and LLVM version 18.1.2)
- os
  - `Linux tjt 5.15.153.1-microsoft-standard-WSL2 #1 SMP Fri Mar 29 23:14:13 UTC 2024 x86_64 GNU/Linux`
- node version:
  - v18.19.0
- @extism/extism vesrion
  - 1.0.3 or 2.0.0-rc3 
- go-pdk
  - 1.0.5

main.go
```golang
package main

import (
	"fmt"
	"github.com/extism/go-pdk"
)

func main() {}

//export hello
func hello() int32 {
	fmt.Println("before", pdk.InputString())
	pdk.OutputString("Hello World")
	fmt.Println("after")
	return 0
}
```
then compiled by `tinygo build -o plugin.wasm -target=wasi main.go`

test.js
```typescript
const createPlugin = require("@extism/extism")

;(async() => {
    const p = await createPlugin("plugin.wasm", {
        useWasi: true,
        enableWasiOutput: true,
    })

    console.log(await p.call("hello", "ppp"))
})()
```
Run with `node test.js`, reuslt:
```
before ppp
after
null  // output block not found
```

Using the exstim cli to call `extism call plugin.wasm hello --wasi` is working expectly

**Comment by bhelx** [comment]
that's surprising! i'll see if i can reproduce today

**Comment by Celend** [comment]
here is the demo https://github.com/Celend/extism-js-sdk-demo , I'm also debugging the lib source :D

**Comment by bhelx** [comment]
hmm, yeah i tried a variety of things. I can't seem to get it to work. I suspect this is happening b/c some languages that compile to WASI have quirks and they don't get consistently initialized. We may need to add in a case for this SDK (which is pretty unique to our others as it's custom to target the W3C wasm interface). 

Let me know if you see anything. I'll also ping @chrisdickinson today and have him take a look. We have a few people on the team who are familiar with these quirks.

**Comment by bhelx** [comment]
Just to report back for anyone else who looks. The linked project uses v2-rc, but it appears to be happening on v1 as well

**Comment by bhelx** [comment]
We are discussing this in Discord. May be an inconsistency with the way that memory is freed when these two projects meet. We'll try to confirm the theory.

**Comment by Celend** [comment]
I saw the discord discussion, remove `extismFree(offset)` https://github.com/extism/go-pdk/blob/7180e57f1b612f1fa769aa5eedde4844f5f6e324/extism_pdk.go#L166 did really worked.

**Comment by bhelx** [comment]
thanks, good to know!

**Comment by nilslice** [comment]
@Celend, this was a recently identified issue, and I missed one place to fix this occurrence. Sorry for that! Will have an update out shortly and will ping here.

**Comment by Celend** [comment]
@nilslice Thanks for UPDATE! but the address probelm still exists, I think it's int64 max value.

If we should reset the memory, reset for every call or some specific frequency?

**Comment by nilslice** [comment]
@Celend - yes, sorry, i should have clarified :) the PR only addresses the output issue. 

as far as the reset frequency goes, I would reset as often as you can tolerate. unless you really need to keep your plugin state alive for a long-running process, it is fast to recreate a plugin and would reset liberally.

**Comment by Celend** [comment]
@nilslice nevermind! but why do not remove the retentions after call? because to avoid memory copy from wasm to host?

**Comment by Celend** [comment]
fixed in 1.0.6
